### PR TITLE
Add dates from the DATS file into the dataset page

### LINKED ISSUE #421: Display the dates DATS fields in the dataset page when information is available
## Purpose

Currently, the dates displayed on the portal for a dataset are based on the Github repo dates.

In the dataset DATS file, there is an optional `dates` field that can be filled by the dataset provider for important dates. These should be displayed in the dataset page when available.

Structure example of the `dates` DATS field:
```
  "dates": [
    {
      "date": "2020-11-30",
      "type": {
        "value": "release date"
      }
    },
    { 
      "date": "2021-02-11",
      "type": {
        "value": "last update date"
      }
    },
    {
      "date": "2011-11-01",
      "type": {
        "value": "start date"
      }
    },
    {
      "date": "2020-03-10",
      "type": {
        "value": "end date"
      }
    }
  ],
```

### PR DESCRIPTION
**Pull request by cmadjar**
This adds the DATS `dates` field in the dataset page under the `more` section to display other dates that may have been stored in the DATS file.

This fixes #421

### CODE REVIEW

**Comment by cmadjar** [comment]
@tkkuehn If you feel like looking at this PR, that would be great. Thank you for offering your help earlier :)

File: app/templates/dataset.html [hunk 1]
```diff
@@ -200,6 +201,16 @@ <h5><strong> {{ data.title }} </strong></h5>
...
+          {% endif %} {% if metadata.dates is not none %}
+          <tr>
+            <th scope="row">Other Dates:</th>
+            <td>
+              {% for key, value in metadata.dates.items() %}
+              <i>{{ key }}:</i> {{ value }}
+              {{" -- " if not loop.last else "" }}
+              {% endfor %}
+            </td>
+          </tr>>
```
Comment (tkkuehn) [inline app/templates/dataset.html:213]: I don't think the extra closing bracket here should be here.

**Review by tkkuehn — COMMENTED** [review]
I think there's one minor typo, but otherwise this looks good to me!

File: app/templates/dataset.html (hunk 1 above)
Comment (cmadjar) [reply app/templates/dataset.html]: ooops. Fixed now. Thank you @tkkuehn !!

**Pull request merged**